url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveLexer.java

```diff
@@ -0,0 +1,15495 @@
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:326:11: ( 'december' ( 's' )? | 'dec' ( DOT )? )
+			int alt38=2;
+			int LA38_0 = input.LA(1);
+			if ( (LA38_0=='d') ) {
+				int LA38_1 = input.LA(2);
+				if ( (LA38_1=='e') ) {
+					int LA38_2 = input.LA(3);
+					if ( (LA38_2=='c') ) {
+						int LA38_3 = input.LA(4);
+						if ( (LA38_3=='e') ) {
+							alt38=1;
+						}
+
+						else {
+							alt38=2;
+						}
+
+					}
+
+					else {
+						int nvaeMark = input.mark();
+						try {
+							for (int nvaeConsume = 0; nvaeConsume < 3 - 1; nvaeConsume++) {
+								input.consume();
+							}
+							NoViableAltException nvae =
+								new NoViableAltException("", 38, 2, input);
+							throw nvae;
+						} finally {
+							input.rewind(nvaeMark);
+						}
+					}
+
+				}
+
+				else {
+					int nvaeMark = input.mark();
+					try {
+						input.consume();
+						NoViableAltException nvae =
+							new NoViableAltException("", 38, 1, input);
+						throw nvae;
+					} finally {
+						input.rewind(nvaeMark);
+					}
+				}
+
+			}
+
+			else {
+				NoViableAltException nvae =
+					new NoViableAltException("", 38, 0, input);
+				throw nvae;
+			}
+
+			switch (alt38) {
+				case 1 :
+					// HiveLexer.g:326:13: 'december' ( 's' )?
+					{
+					match("december"); 
+
+					// HiveLexer.g:326:25: ( 's' )?
+					int alt36=2;
+					int LA36_0 = input.LA(1);
+					if ( (LA36_0=='s') ) {
+						alt36=1;
+					}
+					switch (alt36) {
+						case 1 :
+							// HiveLexer.g:326:25: 's'
+							{
+							match('s'); 
+							}
+							break;
+
+					}
+
+					}
+					break;
+				case 2 :
+					// HiveLexer.g:326:33: 'dec' ( DOT )?
+					{
+					match("dec"); 
+
+					// HiveLexer.g:326:39: ( DOT )?
+					int alt37=2;
+					int LA37_0 = input.LA(1);
+					if ( (LA37_0=='.') ) {
+						alt37=1;
+					}
+					switch (alt37) {
+						case 1 :
+							// HiveLexer.g:
+							{
+							if ( input.LA(1)=='.' ) {
+								input.consume();
+							}
+							else {
+								MismatchedSetException mse = new MismatchedSetException(null,input);
+								recover(mse);
+								throw mse;
+							}
+							}
+							break;
+
+					}
+
+					}
+					break;
+
+			}
+			state.type = _type;
+			state.channel = _channel;
+		}
+		finally {
+			// do for sure before leaving
+		}
+	}
+	// $ANTLR end "DECEMBER"
+
+	// $ANTLR start "SUNDAY"
+	public final void mSUNDAY() throws RecognitionException {
+		try {
+			int _type = SUNDAY;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:328:11: ( 'sunday' ( 's' )? | 'sun' ( DOT )? | 'suns' ( DOT )? )
+			int alt42=3;
+			int LA42_0 = input.LA(1);
+			if ( (LA42_0=='s') ) {
+				int LA42_1 = input.LA(2);
+				if ( (LA42_1=='u') ) {
+					int LA42_2 = input.LA(3);
+					if ( (LA42_2=='n') ) {
+						switch ( input.LA(4) ) {
+						case 'd':
+							{
+							alt42=1;
+							}
+							break;
+						case 's':
+							{
+							alt42=3;
+							}
+							break;
+						default:
+							alt42=2;
+						}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 7681:7694, 7841:7854, 8560:8573.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.